repose-aggregator/components/filters/cors/src/main/scala/org/openrepose/filters/cors/CorsFilter.scala

```diff
@@ -0,0 +1,180 @@
+              httpServletResponse.setHeader(CommonHttpHeader.ACCESS_CONTROL_ALLOW_HEADERS, _)}
+            getValidMethodsForResource(httpServletRequest.getRequestURI).foreach {
+              httpServletResponse.addHeader(CommonHttpHeader.ACCESS_CONTROL_ALLOW_METHODS, _)}
+            httpServletResponse.setStatus(HttpServletResponse.SC_OK)
+          case ActualRequest =>
+            filterChain.doFilter(httpServletRequest, httpServletResponse)
+            httpServletResponse.setHeader(CommonHttpHeader.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true")
+            httpServletResponse.setHeader(CommonHttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN, origin)
+
+            // clone the list of header names so we can add headers while we iterate through it
+            (List() ++ httpServletResponse.getHeaderNames.asScala).foreach {
+              httpServletResponse.addHeader(CommonHttpHeader.ACCESS_CONTROL_EXPOSE_HEADERS, _)
+            }
+        }
+      case OriginNotAllowed =>
+        httpServletResponse.setHeader(CommonHttpHeader.ORIGIN, "null")
+        httpServletResponse.setStatus(HttpServletResponse.SC_FORBIDDEN)
+      case MethodNotAllowed =>
+        httpServletResponse.setHeader(CommonHttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN, origin)
+        httpServletResponse.setStatus(HttpServletResponse.SC_FORBIDDEN)
+    }
+
+    // always add the Vary header
+    httpServletResponse.addHeader(CommonHttpHeader.VARY, CommonHttpHeader.ORIGIN)
+    if (isOptions) {
+      httpServletResponse.addHeader(CommonHttpHeader.VARY, CommonHttpHeader.ACCESS_CONTROL_REQUEST_HEADERS)
+      httpServletResponse.addHeader(CommonHttpHeader.VARY, CommonHttpHeader.ACCESS_CONTROL_REQUEST_METHOD)
+    }
+  }
+
+  override def destroy(): Unit = {
+    logger.trace("CORS filter destroying...")
+    configurationService.unsubscribeFrom(configurationFile, this.asInstanceOf[UpdateListener[_]])
+    logger.trace("CORS filter destroyed.")
+  }
+
+  override def configurationUpdated(config: CorsConfig): Unit = {
+    allowedOrigins = config.getAllowedOrigins.getOrigin.asScala.map { origin =>
+      if (origin.isRegex) origin.getValue.r else Pattern.quote(origin.getValue).r
+    }
+
+    allowedMethods = Option(config.getAllowedMethods).map(_.getMethod.asScala).getOrElse(List())
+
+    resources = Option(config.getResources)
+      .map(_.getResource.asScala).getOrElse(List())
+      .map { configResource =>
+      Resource(configResource.getPath.r,
+        Option(configResource.getAllowedMethods).map(_.getMethod.asScala).getOrElse(List()))}
+
+    initialized = true
+  }
+
+  override def isInitialized: Boolean = initialized
+
+  def isOriginAllowed(requestOrigin: String): Boolean = allowedOrigins.exists(_.findFirstIn(requestOrigin).isDefined)
+
+  def getValidMethodsForResource(path: String): Iterable[String] = {
+    val resource = resources.find(_.path.findFirstIn(path).isDefined)
+    logger.debug(s"Matched path $path with configured resource: $resource")
```

Comment:
I updated it to match on the output of the .find(...) and used the existing log message for success and a new message for a failure.  It looks cleaner now.  Good stuff.